Update ZHA config flow Zigbee radio description

Hedda opened the pull request:
## Breaking change
No

## Proposed change
Update ZHA config flow Zigbee radio descriptions for "radio type" user get to pick to be used for ZHA.

 ezsp = Silicon Labs Ember based radios with EmberZNet Zigbee firmware
 deconz = dresden elektronik ConBee and RaspBee based radios with deCONZ Zigbee firmware
 ti_cc = Texas Instruments CC253x/CC26x2/CC13x2 based radios with Z-Stack firmware
 zigate = ZiGate USB-TTL, PiZiGate, and WiFi based Zigbee radios with ZiGate firmware
 xbee = Digi XBee Series 2 and 3 based radios with XBee Zigbee firmware

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
n/a

## Additional information
n/a

## Checklist
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

Hedda committed:
Update ZHA config flow Zigbee radio description

Update ZHA config flow Zigbee radio descriptions to match docs https://github.com/home-assistant/home-assistant.io/pull/13437

Adminiuga commented:
See https://github.com/home-assistant/core/pull/35397#discussion_r422409616 comment chain

Adminiuga commented:
And internal name of the radio is none  of  user concern  so I don't see much point  making it lower case just  because it used to be  so.

Hedda committed:
Make protocol plus model names more readable for end-users

IMHO, this makes protocol used plus common model names both more readable and pretty.

Also no longer use all lower case letters for what is just the internal names of radios.

Hedda commented:
OK How about:

* "EZSP = Silicon Labs EmberZNet protocol (ex; Elelabs, HUSBZB-1, Telegesis)"
* "deCONZ = dresden elektronik deCONZ protocol (ex; ConBee I/II, RaspBee I/II)"
* "TI_CC = Texas Instruments Z-Stack ZNP protocol (ex; CC253x, CC26x2, CC13x2)"
* "ZiGate = ZiGate Serial protocol (ZiGate WiFi, ZiGate USB-TTL, PiZiGate)"
* "XBee = Digi XBee ZB Coordinator Firmware protocol (Digi XBee Series 2, 2C, 3)"

PS: Yes I can confirm that I have CDO (it is like OCD but all the letters are in alphabetical order, as they should be).

Adminiuga commented on homeassistant/components/zha/core/const.py:
```diff
@@ -157,20 +157,23 @@ class RadioType(enum.Enum):
 
     ezsp = (
-        "ESZP: HUSBZB-1, Elelabs, Telegesis, Silabs EmberZNet protocol",
+        "EZSP = Silicon Labs EmberZNet protocol (ex; Elelabs, HUSBZB-1, Telegesis)",
```
IMO should be a colon in `ex:` since the models are closely related to the previous clause

Hedda replied:
OK done

Adminiuga commented on homeassistant/components/zha/core/const.py:
```diff
@@ -157,20 +157,23 @@ class RadioType(enum.Enum):
     )
     deconz = (
-        "Conbee, Conbee II, RaspBee radios from dresden elektronik",
+        "deCONZ = dresden elektronik deCONZ protocol (ex; ConBee I/II, RaspBee I/II)",
```
There are only ConBees and RaspBees, so for this one could remove ex; completely
```suggestion
        "deCONZ = dresden elektronik deCONZ protocol: ConBee I/II, RaspBee I/II",
```

Adminiuga commented on homeassistant/components/zha/core/const.py:
```diff
@@ -157,20 +157,23 @@ class RadioType(enum.Enum):
     )
     ti_cc = (
-        "TI_CC: CC2531, CC2530, CC2652R, CC1352 etc, Texas Instruments ZNP protocol",
+        "TI_CC = : Texas Instruments Z-Stack ZNP protocol (ex; CC253x, CC26x2, CC13x2)",
```
```suggestion
        "TI_CC = Texas Instruments Z-Stack ZNP protocol (ex: CC253x, CC26x2, CC13x2)",
```

Adminiuga commented on homeassistant/components/zha/core/const.py:
```diff
@@ -157,20 +157,23 @@ class RadioType(enum.Enum):
     )
-    zigate = "ZiGate Radio", zigpy_zigate.zigbee.application.ControllerApplication
+    zigate = (
+        "ZiGate = ZiGate Serial protocol (ZiGate WiFi, ZiGate USB-TTL, PiZiGate)",
```
```suggestion
        "ZiGate = ZiGate Serial protocol: PiZiGate, ZiGate USB-TTL, ZiGate WiFi",
```
let's just have them in alphabetical order

Adminiuga replied:
Also, this is for radio type, do we necessarily have to include "serial protocol" words? `ZiGate Radio` won't suffice?

Adminiuga commented on homeassistant/components/zha/core/const.py:
```diff
@@ -157,20 +157,23 @@ class RadioType(enum.Enum):
+    )
     xbee = (
-        "Digi XBee S2C, XBee 3 radios",
+        "XBee = Digi XBee ZB Coordinator Firmware protocol (Digi XBee Series 2, 2C, 3)",
```
Same as with ZiGate. Digi XBee modules? or Digi XBee radio?

Hedda replied:
OK done that too

Hedda committed:
Update homeassistant/components/zha/core/const.py

Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

Hedda committed:
Update homeassistant/components/zha/core/const.py

Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

Hedda committed:
Update homeassistant/components/zha/core/const.py

Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

Hedda committed:
Zigbee radio description simplified and examples  listed in alphabetical order

Zigbee radio description simplified and examples  listed in alphabetical order

Hedda commented:
Though maybe "TI_CC" should not have an underscore either? So just "TI CC" instead?

"TI CC = Texas Instruments Z-Stack ZNP protocol (ex; CC253x, CC26x2, CC13x2)" or?

Hedda commented:
FYI, submitted matching PR for ZHA docs https://github.com/home-assistant/home-assistant.io/pull/13695
